File: algorithm/HDU-ACM/ambition0109/1018.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-05-19 12:05:59
////Compiler: Visual C
//////////////////////////////////////////////////////
////Problem ID: 1018
////Problem Title: 
////Run result: Accept
////Run time:390MS
////Run memory:228KB
//////////////////System Comment End//////////////////
#include<stdio.h>
#include<math.h>
main()
{
	int n,i,t;
	double d;
	for(scanf("%d",&t);t>0;t--)
	{
		scanf("%d",&n);
		for(i=1,d=0;i<=n;i++)
			d+=log10(i);
		printf("%d\n",(int)d+1);
	}
}

File: algorithm/杭电ACM/ambition0109/1018.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-11 22:27:36
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1018
////Problem Title: 
////Run result: Accept
////Run time:500MS
////Run memory:6072KB
//////////////////System Comment End//////////////////
import java.util.Scanner;

public class Main {
	static public void main(String args[]){
		Scanner in=new Scanner(System.in);
		int t=in.nextInt();
		while(t--!=0){
			int n=in.nextInt();
			double ans=0.0;
			for(int i=1;i<=n;i++){
				ans+=Math.log10((double)i);
			}
			System.out.println(((int)ans+1));
		}
	} 
}
